### ISSUE TITLE
I/O emulation

### ISSUE THREAD

**Issue by mingtaiha** [radical-cybertools/radical.synapse#15]
I know Synapse can profile how much computation an application performs by recording the number of cycles used, and can also emulate the same number of cycles. Can Synapse do the same for I/O reads and writes?

**Comment by andre-merzky** [comment]
Yes!  The caveat here is that we only sample I/O, and have no ability to look into the size of individual I/O requests (read/write buffer sizes).  Those will, however, significantly influence overall I/O performance.

I have been playing around with watching I/O block sizes in the past, but that is not in a usable state, and not in the mainline code.  Happy to discuss if and how this could be revived and advanced, if needed.
